src/layerPart.cpp

```diff
@@ -65,22 +65,60 @@ void createLayerWithParts(const Settings& settings, SliceLayer& storageLayer, Sl
         result = layer->polygons_.splitIntoParts(union_layers || union_all_remove_holes);
     }
 
-    for (auto& part : result)
+    for (auto& main_part : result)
     {
-        storageLayer.parts.emplace_back();
-        if (part.empty())
+        std::vector<std::pair<SliceLayerPart::WallExposedType, std::vector<SingleShape>>> parts_by_type = {
+            { SliceLayerPart::WallExposedType::BOTTOM_0, bottom_parts.splitIntoParts() },
+            { SliceLayerPart::WallExposedType::TOP, top_parts.difference(bottom_parts).splitIntoParts() },
+            { SliceLayerPart::WallExposedType::SIDE_ONLY, main_part.difference(bottom_parts).difference(top_parts).splitIntoParts() },
+        };
+
+        for (auto& [wall_exposed, parts] : parts_by_type)
         {
-            continue;
+            for (auto& part : parts)
+            {
+                storageLayer.parts.emplace_back();
+                if (part.empty())
+                {
+                    continue;
+                }
+                auto& back_part = storageLayer.parts.back();
+                back_part.wall_exposed = wall_exposed;
+                back_part.outline = part;
+                back_part.boundaryBox.calculate(back_part.outline);
+                if (back_part.outline.empty())
+                {
+                    storageLayer.parts.pop_back();
+                }
+            }
         }
-        storageLayer.parts.back().outline = part;
-        storageLayer.parts.back().boundaryBox.calculate(storageLayer.parts.back().outline);
-        if (storageLayer.parts.back().outline.empty())
+    }
+}
+
+Shape getTopOrBottom(int direction, const std::string& setting_name, size_t layer_nr, const std::vector<SlicerLayer>& slayers, const Settings& settings)
+{
+    auto result = Shape();
```

Comment:
I'm probably old-style and not a huge fan of `auto`, but I have a slight preference for this:
```suggestion
    Shape result;
```

Reply:
Yeah makes sense. I mostly use `auto` when the type-name would otherwise double, but here you can just use the short notation instead.